templates/lobby.html

```diff
@@ -652,7 +653,7 @@
     };
 
     // necessary for mousemove to not use the previous exit coordinates.
-    drawingBoard.onmouseenter = function (e) {
+    window.onmouseenter = function (e) {
```

Comment:
I assume we need `onmouseexit` to be clean, since as it is right now, we'd technically send invalid events when moving the mouse outside of the canvas, or am i seeing that the wrong way?

**EDIT** Draw-line handles this, but this won't do for third party clients.

Reply:
Technically the mouse enter events can be ignored when catching the window events, since that is necessary to have lines not cut off at the edges. I'll try implementing it with onmouseexit, since I also think it would be cleaner.

Reply:
onmouseenter fires too late, leading to the line starting somewhere on the canvas instead of the edges.

Reply:
Do the bound checks somewhere before sending the events then. But avoid doing them twice. Alternatively, we just add them to the backend :shrug: This way no client could make mistakes.

Reply:
Right now, it's not easily possible to avoid lines starting or ending outside the canvas, since the mouse is just not polled often enough.  
It would be possible to calculate the intersection with the canvas and rewrite start or end point. Do you want me to do that? I don't know if you want to guarantee that lines have to start on the canvas for all clients

Reply:
Some of those algorithms could be implemented: https://en.wikipedia.org/wiki/Line_clipping

Reply:
alternatively we could just add a check on mousemove, e.g. when it attempts to draw a line either way, right?

Reply:
I added another change doing this. It now automatically cuts coordinates to the nearest canvas coordinate, which sometimes produces sharp turns, but looks fine most of the time.